mbed_build/_internal/config/config_layer.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Layered approach to applying config sources to Config class."""
+from dataclasses import dataclass
+from typing import Iterable, List
+
+from mbed_build._internal.config.config_action import ConfigAction
+from mbed_build._internal.config.config_source import ConfigSource
+
+# Fix circular dependency problem
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mbed_build._internal.config.config import Config
+
+
+@dataclass
+class ConfigLayer:
+    """ConfigLayer is a simple container of actions which can be used to build Config."""
```

Comment:
I would give an example here of what you mean by layers. It's quite abstract at the moment, even with the description.

Reply:
I've imporoved the docstring slightly, better describing the responsibilities of the layer. This spans over many commits. :P